cluster-autoscaler/cloudprovider/azure/azure_config.go

```diff
@@ -212,6 +218,15 @@ func BuildAzureConfig(configReader io.Reader) (*Config, error) {
 			}
 		}
 
+		if enableDynamicInstanceList := os.Getenv("AZURE_ENABLE_DYNAMIC_INSTANCE_LIST"); enableDynamicInstanceList != "" {
```

Comment:
nit: can we have this as `AUTOSCALER_ENABLE_DYNAMIC_INSTANCE_LIST` and prefix the json and yaml keys with `autoscalerEnableDynamicInstanceList`.

I presume a lot of ppl may have that secret mounted to both CA and cloudprovider so it might be worth having a prefix.